src/coord/src/coord.rs

```diff
@@ -550,12 +590,8 @@ impl Coordinator {
     // backward). This downgrades the capabilities of all tables, which means that
     // all tables can no longer produce new data before this timestamp.
     fn advance_local_inputs(&mut self) {
-        let mut next_ts = self.get_ts();
         self.need_advance = false;
-        if next_ts <= self.read_lower_bound {
-            next_ts = self.read_lower_bound + 1;
-        }
-        // These can occasionally be equal, so ignore the update in that case.
+        let next_ts = self.system_timeline().get_write_ts() + 1;
```

Comment:
This gets `+1` here, and `-1` later on in `maybe_advance`. Is it possible that the `write_ts` itself is the right thing to track? The various potential off-by-one, could be zero, could be intmax things worry me.

Reply:
I'm similarly worried. Yes, I think you're right and I was using a timestamp one higher than necessary.

Reply:
I took out the `+1` which resulted in availability problems since the timestamp wasn't closed. Queries using system tables would take tens of seconds, blocked on something waiting to close off the timestamp. This was causing intermittent test failures. I've restored the `+1`.